src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/adapter/WebClientUtils.kt

```diff
@@ -102,6 +103,18 @@ class WebClientUtils(private val client: WebClient) {
       .bodyToMono(elementClass)
       .block()!!
 
+  fun <T : Any> postWithResponse(uri: String, body: Any, elementClass: Class<T>, status: HttpStatus, newException: Exception): T =
+    try {
+      client.post()
+        .uri(uri)
+        .bodyValue(body)
+        .retrieve()
+        .bodyToMono(elementClass)
+        .block()!!
+    } catch (e: WebClientResponseException) {
+      throw if (e.statusCode.equals(status)) newException else e
```

Comment:
This method is the same as the postWithResponse method but with the additional exception code